gosec showing various errors.

I dont know if this is checked somewhere else because there is no makefile, but i found the following.



``` 
go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest

golangci-lint run -E gosec -E sqlclosecheck ./... --skip-dirs-use-default
sql/internal/sqlx/twin.go:9:2: G501: Blocklisted import crypto/md5: weak cryptographic primitive (gosec)
        "crypto/md5"
        ^
sql/internal/sqlx/twin.go:113:64: G401: Use of weak cryptographic primitive (gosec)
        return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-md5.Size*2], md5.Sum([]byte(twin)))
                                                                      ^
sql/sqlite/migrate.go:415:19: Close should use defer (sqlclosecheck)
                err = rows.Close()
                                ^
cmd/action/fmt_test.go:144:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
                err = os.WriteFile(file, []byte(contents), 0644)
                      ^
sql/migrate/migrate.go:263:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return os.WriteFile(filepath.Join(d.dir, name), b, 0644)
               ^
cmd/action/internal/update/update.go:149:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fileLocation(path), b, 0644)
               ^
make: *** [lint] Error 1

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,7 +14,7 @@ jobs:
         uses: golangci/golangci-lint-action@v2.5.2
         with:
           version: v1.42.1
-          args: --verbose --enable revive -E gosec
+          args: --verbose
   generate-cmp:
     runs-on: ubuntu-latest
     steps:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,7 +14,7 @@ jobs:
         uses: golangci/golangci-lint-action@v2.5.2
         with:
           version: v1.42.1
-          args: --verbose --enable revive
+          args: --verbose --enable revive -E gosec
   generate-cmp:
     runs-on: ubuntu-latest
     steps:

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -4,3 +4,9 @@ run:
 issues:
   include:
     - EXC0012
+
+linters:
+  disable-all: true
+  enable:
+    - gosec
+    - revive

--- a/cmd/action/fmt_test.go
+++ b/cmd/action/fmt_test.go
@@ -141,7 +141,7 @@ func setupFmtTest(t *testing.T, inputDir map[string]string) string {
 	})
 	for name, contents := range inputDir {
 		file := path.Join(dir, name)
-		err = os.WriteFile(file, []byte(contents), 0644)
+		err = os.WriteFile(file, []byte(contents), 0600)
 	}
 	require.NoError(t, err)
 	return dir

--- a/cmd/action/internal/update/update.go
+++ b/cmd/action/internal/update/update.go
@@ -146,7 +146,7 @@ func saveStore(path string, l LatestRelease, t time.Time) error {
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(fileLocation(path), b, 0644)
+	return ioutil.WriteFile(fileLocation(path), b, 0600)
 }
 
 func fileLocation(p string) string {

--- a/sql/internal/sqlx/twin.go
+++ b/sql/internal/sqlx/twin.go
@@ -110,6 +110,6 @@ func (t *TwinDriver) formatName(name string) string {
 	if t.MaxNameLen == 0 || len(twin) <= t.MaxNameLen {
 		return twin
 	}
-	hasher := fnv.New128()
-	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-hasher.Size()*2], hasher.Sum([]byte(twin)))
+	h := fnv.New128()
+	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-h.Size()*2], h.Sum([]byte(twin)))
 }

--- a/sql/internal/sqlx/twin.go
+++ b/sql/internal/sqlx/twin.go
@@ -14,8 +14,6 @@ import (
 	"ariga.io/atlas/sql/schema"
 )
 
-var hasher = fnv.New128()
-
 // TwinDriver is a driver that provides additional functionality
 // to interact with the twin/dev database.
 type TwinDriver struct {
@@ -112,5 +110,6 @@ func (t *TwinDriver) formatName(name string) string {
 	if t.MaxNameLen == 0 || len(twin) <= t.MaxNameLen {
 		return twin
 	}
+	hasher := fnv.New128()
 	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-hasher.Size()*2], hasher.Sum([]byte(twin)))
 }

--- a/sql/internal/sqlx/twin.go
+++ b/sql/internal/sqlx/twin.go
@@ -6,13 +6,15 @@ package sqlx
 
 import (
 	"context"
-	"crypto/md5"
 	"fmt"
+	"hash/fnv"
 	"time"
 
 	"ariga.io/atlas/sql/migrate"
 	"ariga.io/atlas/sql/schema"
 )
+
+var hasher = fnv.New128()
 
 // TwinDriver is a driver that provides additional functionality
 // to interact with the twin/dev database.
@@ -110,5 +112,5 @@ func (t *TwinDriver) formatName(name string) string {
 	if t.MaxNameLen == 0 || len(twin) <= t.MaxNameLen {
 		return twin
 	}
-	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-md5.Size*2], md5.Sum([]byte(twin)))
+	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-1-hasher.Size()*2], hasher.Sum([]byte(twin)))
 }

--- a/sql/migrate/migrate.go
+++ b/sql/migrate/migrate.go
@@ -260,7 +260,7 @@ func (d *LocalDir) Open(name string) (fs.File, error) {
 
 // WriteFile implements Dir.WriteFile.
 func (d *LocalDir) WriteFile(name string, b []byte) error {
-	return os.WriteFile(filepath.Join(d.dir, name), b, 0644)
+	return os.WriteFile(filepath.Join(d.dir, name), b, 0600)
 }
 
 var (

